scrapy/spiders/__init__.py

```diff
@@ -89,6 +91,27 @@ def close(spider, reason):
         if callable(closed):
             return closed(reason)
 
+    @staticmethod
+    def idle(spider):
+        idled = getattr(spider, 'idled', None)
+        if callable(idled):
+            cb_response = idled()
+            if cb_response is None:
+                return
+            if isinstance(cb_response, Request):
+                #spider.crawler.engine.crawl(cb_response, spider)
+                spider.crawler.engine.schedule(cb_response, spider)
+                spider.crawler.engine.slot.nextcall.schedule()
+                raise DontCloseSpider
+            if isinstance(cb_response, collections.Iterable):
+                for request in cb_response:
+                    #spider.crawler.engine.crawl(request, spider)
+                    spider.crawler.engine.schedule(request, spider)
+                spider.crawler.engine.slot.nextcall.schedule()
```

Comment:
Which is right? calling `engine.crawl()` vs `engine.schedule()`?
Calling schedule() without slot.nextcall.schedule() will cause noticeable delays before the next run. But maybe it should be like that